odoo/addons/base/models/ir_property.py

```diff
@@ -87,12 +85,43 @@ def _update_values(self, values):
 
     @api.multi
     def write(self, values):
-        return super(Property, self).write(self._update_values(values))
+        # if any of the records we're writing on has a res_id=False *or*
+        # we're writing a res_id=False on any record
+        default_set = False
+        if self._ids:
+            self.env.cr.execute(
+                'SELECT EXISTS (SELECT 1 FROM ir_property WHERE id in %s AND res_id IS NULL)', [self._ids])
+            default_set = self.env.cr.rowcount == 1 or any(
```

Comment:
`any` waits to process all items to return a result then if you have the result required in the first item it is slower than a break-early

`python3.5 -m timeit -s "items = range(100)" "result=any([item == 0 for item in items])"`
`100000 loops, best of 3: 6.05 usec per loop`

`python3.5 -m timeit -s "items = range(100)" "result=False" "for item in items:" "  if item==0:" "    result=True;break"`
`10000000 loops, best of 3: 0.11 usec per loop`

Other way you can use `False in dict.values` for this case.

`python3.5 -m timeit -s "items = dict((item,item) for item in range(100))" "result = False in items.values()"`
`10000000 loops, best of 3: 0.17 usec per loop`

Reply:
> any waits to process all items to return a result

Only because you used a `any` on a list comprehension in your example, so the list is generated completely before being processed by any. If you use a generator `any(item == 0 for item in items)`, it will stop at the first item found before generating the whole list. (it might still be slower for other reasons though, didn't checked)

Reply:
> `any` waits to process all items to return a result

It doesn't, that's the list comprehension doing that. `any` will return as soon as it gets a truthy item.